tests/foreman/longrun/test_oscap.py

```diff
@@ -110,71 +136,80 @@ def activation_key(module_target_sat, module_org, lifecycle_env, content_view):
+    if result.status != 0:
+        raise ContentHostError('Failed to install scap-security-guide')
+    content_dir = '/usr/share/xml/scap/ssg/content/'
+    result = target_sat.execute(
+        f'''sshpass -p "{settings.server.ssh_password}" scp -o StrictHostKeyChecking=no root@{contenthost.hostname}:{content_dir}* {content_dir}'''
+    )
+    assert result.status == 0, f'failed to copy content data from {contenthost.hostname}'
+    target_sat.cli.Scapcontent.bulk_upload(
+        {'type': 'directory', 'directory': content_dir, 'organization-id': org.id}
+    )
 
-    :setup: scap content, scap policy, host group
 
-    :steps:
+def get_existing_content(target_sat):
+    """List existing scap content"""
+    existing_content = target_sat.cli.Scapcontent.list()
+    logger.debug(f'Listing existing content on Satellite: {existing_content}')
+    return [content['title'] for content in existing_content]
 
-        1. Create a valid scap content
-        2. Import Ansible role theforeman.foreman_scap_client
-        3. Import Ansible Variables needed for the role
-        4. Create a scap policy with ansible as deploy option
-        5. Associate the policy with a hostgroup
-        6. Provision a host using the hostgroup
-        7. Configure REX and associate the Ansible role to created host
-        8. Play roles for the host
 
-    :expectedresults: REX job should be success and ARF report should be sent to satellite
+def find_content_to_update(target_sat, module_org, distro, contenthost):
+    """find out what content to update based on existing content"""
+    selected_content = OSCAP_DEFAULT_CONTENT[f'{distro}_content']
+    existing_content_names = get_existing_content(target_sat)
+    # the content name depends on upload style (default vs file upload)
+    if (selected_content not in existing_content_names) and (
+        f'{distro} content' not in existing_content_names
+    ):
+        upload_scap_content_for_host(target_sat, contenthost, module_org)
 
-    :BZ: 1716307, 1992229
+    # re-list after content upload
+    existing_content_names = get_existing_content(target_sat)
+    # find out content name for client version
+    assert (selected_content in existing_content_names) or (
+        f'{distro} content' in existing_content_names
+    ), f'scap content not found for {distro}, existing content: {existing_content_names}'
+    return selected_content if (selected_content in existing_content_names) else f'{distro} content'
 
-    :customerscenario: true
 
-    :CaseImportance: Critical
-    """
-    if distro == 'rhel7':
-        rhel_repo = settings.repos.rhel7_os
-        profile = OSCAP_PROFILE['security7']
-    else:
-        rhel_repo = settings.repos.rhel8_os
-        profile = OSCAP_PROFILE['ospp8']
+@pytest.fixture
+def scap_prerequisites(module_org, default_proxy, target_sat):
+    # TODO: add support for RHEL9 (it doesn't have scap content in Sat by default) and parametrize distro
```

Comment:
IIRC the problem is that we install `scap-security-guide` on Satellite, but if that's on RHEL 8 then it doesn't have RHEL 9 content. AFAIK this is why we upload the content from the content host (which you do in `upload_scap_content_for_host`). Though if the Satellite is on RHEL 9 then it should have the content available.

Would it be better to match the distro to the Satellite distro?

Reply:
eh, further complication, this part got in with https://github.com/SatelliteQE/robottelo/pull/15491 while this PR was open, I just rearranged stuff after I pulled in changes. Clearly we should consolidate the approches form thes PRs, but I don't want to touch it within this PR.

Reply:
That's fair. I'd just ask you to address https://github.com/SatelliteQE/robottelo/pull/15531/commits/13fef985e381600764166c3bdf17e3ff981be9fd#r1691678387 since that does feel in line with this PR (you're introducing `profiles` in this PR).

Reply:
@ekohl ah that's right, comitted!